chef/cookbooks/provisioner/recipes/bootdisk.rb

```diff
@@ -78,17 +79,38 @@
         node[:crowbar_wall][:boot_device] = "disk/by-id/#{bootdisk}"
       end
     end
+
     # If we still have a /dev/disk/by-path name in boot_device here and the
     # devices looks like a virtio device it is safer to use the base device
     # name for bootdisk instead of the by-path name. According to:
     #  https://www.mail-archive.com/systemd-devel@lists.freedesktop.org/msg20607.html
     # the by-path names for virtio device are not stable and don't even exist
-    # anymore with newer udev releases, e.g. on SLES12
-    if File.dirname(node[:crowbar_wall][:boot_device]) == "disk/by-path" and
-        File.basename(node[:crowbar_wall][:boot_device]) =~ /-virtio-/
+    # anymore with newer udev releases
+    # This is only for SLE_11_SP3 sleshammer. On SLE_12 based image
+    # directory "disk/by-path" with virtio disks driver doesn't exist
+    if !node[:crowbar_wall][:boot_device].nil? and
+         File.dirname(node[:crowbar_wall][:boot_device]) == "disk/by-path" and
```

Comment:
Style/AndOr: Use `&&` instead of `and`.<br>Style/MultilineOperationIndentation: Use 4 (not 5) spaces for indenting a condition in an `if` statement spanning multiple lines.